packages/machine/src/vm.ts

```diff
@@ -173,10 +176,12 @@ export class CounterfactualVM implements Observable {
     status: cf.node.ResponseStatus
   ) {
     if (!execution.action.isAckSide) {
-      this.responseHandler.sendResponse(
-        new cf.node.Response(execution.action.requestId, status)
-      );
+      return;
     }
+
+    this.responseHandler.sendResponse(
+      new cf.node.Response(execution.action.requestId, status)
+    );
```

Comment:
Why the change?

Reply:
It's usually not good practice to have a single if-condition wrapping the entirety of a function. If for any reason, a function must abort its execution, it's preferred to do so as quickly as possible.

Reply:
Shouldn't the if clause be reversed though?

Reply:
Yeah, you're right on that, forgot to remove the `!`.